lib/core/CLogger.cc

```diff
@@ -167,6 +170,21 @@ void CLogger::fatal() {
     throw std::runtime_error("Ml Fatal Exception");
 }
 
+void CLogger::fatalErrorHandler(const TFatalErrorHandler& handler) {
+    // This is purposely not thread safe. This is only meant to be called once from
+    // the main thread, typically from main of an executable or in single threaded
+    // test code.
+    m_FatalErrorHandler = handler;
+}
+
+const CLogger::TFatalErrorHandler& CLogger::fatalErrorHandler() const {
+    return m_FatalErrorHandler;
+}
+
+void CLogger::handleFatal(std::string message) {
+    m_FatalErrorHandler(message);
```

Comment:
This is double copying the string at the moment.  If you don't like my suggestion to make the error handler function take a `const std::string&` then this could move `message`.